src/ucs/profile/profile_defs.h

```diff
@@ -55,22 +66,42 @@ typedef enum {
  *    < ucs_profile_record_t > * ucs_profile_thread_header_t::num_records
  *
  * ] * ucs_profile_thread_header_t::num_threads
+ * <env variables string>
  */
 
 
+typedef struct {
+    uint64_t offset;
+    uint64_t size;
+} UCS_S_PACKED ucs_profile_block_t;
+
+
+#define UCS_PROFILE_WRITE_BLOCK(_fd, _profile_block, _func, ...) \
+    ({ \
+        ucs_status_t _status; \
+        _profile_block.offset = lseek(_fd, 0, SEEK_CUR); \
+        _status               = _func(_fd, __VA_ARGS__); \
+        _profile_block.size = lseek(_fd, 0, SEEK_CUR) - _profile_block.offset; \
+        _status; \
+    })
+
 /**
  * Profile output file header
  */
 typedef struct ucs_profile_header {
-    uint32_t                 version;       /**< File format version */
-    char                     ucs_path[1024];/**< UCX library path*/
-    char                     cmdline[1024]; /**< Command line */
-    char                     hostname[64];  /**< Host name */
-    uint32_t                 pid;           /**< Process ID */
-    uint32_t                 mode;          /**< Bitmask of profiling modes */
-    uint32_t                 num_locations; /**< Number of locations in the file */
-    uint32_t                 num_threads;   /**< Number of threads in the file */
-    uint64_t                 one_second;    /**< How much time is one second on the sampled machine */
+    uint32_t            version;         /**< File format version */
+    uint32_t            feature_flags;   /**< Feature flags bitmask */
+    char                ucs_path[1024];  /**< UCX library path*/
+    char                cmdline[1024];   /**< Command line */
+    char                hostname[64];    /**< Host name */
+    uint32_t            pid;             /**< Process ID */
+    uint32_t            mode;            /**< Bitmask of profiling modes */
+    ucs_profile_block_t locations_block; /**< Size and offset of locations*/
+    ucs_profile_block_t threads_block;   /**< Size and offset of threads*/
+    ucs_profile_block_t env_block;       /**< Size and offset of environment variables string*/
+    uint32_t num_locations;              /**< Number of locations in the file */
+    uint32_t num_threads;                /**< Number of threads in the file */
```

Comment:
1. why need both size and count of threads and locations? I think we can calc the count from block size field
2. if really needed consider moving it to block header structure